torch/csrc/jit/tensorexpr/analysis.h

```diff
@@ -67,6 +68,31 @@ class VarFinder : public IRVisitor {
   std::unordered_set<const Var*> vars_;
 };
 
+// A class that analyzes the given program relevant for Block backend
+// It creates a map of multi dim buffers and their flat verions
+class CreateBufferMap : public IRVisitor {
+ public:
+  const std::unordered_map<std::string, const Buf*>& getBufferMap() const {
+    return map_input_to_tensor_bufs_;
+  }
+
+ private:
+  void visit(const Store* v) override {
+    auto load_node = dynamic_cast<const Load*>(v->value());
+    auto call_node = dynamic_cast<const FunctionCall*>(v->value());
+    if (load_node || call_node) {
+      TORCH_INTERNAL_ASSERT(!(load_node && call_node));
+      auto t_buf = load_node ? load_node->buf() : call_node->tensor()->buf();
+      if (load_node) {
+        map_input_to_tensor_bufs_.emplace(t_buf->name_hint(), v->buf());
```

Comment:
I'm not sure it's a good idea to use 'name_hint' as a key. I think we never tried to guarantee that they were unique.